libraries/APM_Control/AR_AttitudeControl.cpp

```diff
@@ -381,3 +419,13 @@ bool AR_AttitudeControl::get_forward_speed(float &speed) const
     speed = velocity.x*_ahrs.cos_yaw() + velocity.y*_ahrs.sin_yaw();
     return true;
 }
+
+// get latest desired speed recorded during call to get_throttle_out_speed.  For reporting purposes only
+float AR_AttitudeControl::get_desired_speed() const
+{
+    // return zero if no recent calls to speed controller
+    if ((_speed_last_ms == 0) || ((AP_HAL::millis() - _speed_last_ms) > AR_ATTCONTROL_TIMEOUT)) {
+        return 0.0f;
```

Comment:
This doesn't look great; return bool instead?

Again, not a great fan of timeouts in here....